R/test-files.r

```diff
@@ -77,26 +123,42 @@ source_dir <- function(path, pattern = "\\.[rR]$", env = test_env(),
 #' @param path path to file
 #' @param reporter reporter to use
 #' @param env environment in which to execute the tests
-#' @return a data frame of the summary of test results
+#' @return the results as a "testthat_results" (list)
 #' @export
 test_file <- function(path, reporter = "summary", env = test_env()) {
-  current_reporter <- find_reporter(reporter)
+  reporter <- if (is.null(reporter)) NULL else find_reporter(reporter)
+  invisible(.test_file(path, env, reporter, TRUE))
+}
+
+.test_file <- function(path, parent_env, reporter, start_end_reporter) {
+  if (is.null(parent_env)) parent_env <- globalenv()
```

Comment:
It's just a bit weird to use this form of setting a default value, and makes it harder for me to figure out after the fact how it's supposed to be used.